src/plugins/discover/public/application/angular/discover.js

```diff
@@ -73,6 +73,7 @@ import {
   getDefaultQuery,
   search,
 } from '../../../../data/public';
+import { SEARCH_QUERY_LANGUAGE_SETTINGS } from '../../../../data/common';
```

Comment:
No big deal, but I thought it would be recommended to re-export in "public" and import from there in other plugins?

Reply:
++ For legacy platform these imports don't bother me, but for KP plugins we should avoid imports from `/common` until https://github.com/elastic/kibana/issues/59710 is addressed.

Reply:
@lukeelmers @flash1293 The same question I added here: https://github.com/elastic/kibana/pull/64321/files#r425774734 

Also I don't like an idea to reexport all as it from our public static contract. Maybe we can create some namespace for it e.g. `UI_SETTINGS`?

Reply:
@alexwizp As mentioned in https://github.com/elastic/kibana/pull/66040#pullrequestreview-414394234 the solution I would prefer is to only expose a getter in setup and start contracts of the plugin. If that's too big of a requirement for this PR, I'm fine with a namespace, but long-term we should get away from it completely.

Reply:
Thanks @flash1293  --  I had missed that thread on the other PR but the getter approach makes sense to me and is more explicit than relying on the imports of these static consts.

> If that's too big of a requirement for this PR, I'm fine with a namespace, but long-term we should get away from it completely.

I'm fine with namespacing for the time being too -- for two reasons:
1. At least we are still creating a dependency on the data plugin via static imports (which will soon have to become explicit anyway once https://github.com/elastic/kibana/issues/64843 lands)
2. Updating every static import across plugins to make them rely on a runtime service is going to increase the scope of this PR significantly. To me the priority here is migrating these out of the legacy platform.

> If this isn't done is this PR we should open a separate issue to clean that up later

++ I've created an issue to track this: https://github.com/elastic/kibana/issues/67060 

In the meantime, my vote would be to use a `UI_SETTINGS` const as a namespace like @alexwizp suggested.